src/Okta.Sdk/Internal/UserAgentBuilder.cs

```diff
@@ -22,12 +23,15 @@ public sealed class UserAgentBuilder
 
         // Lazy ensures this only runs once and is cached.
         private readonly Lazy<string> _cachedUserAgent;
+        private readonly ILogger _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="UserAgentBuilder"/> class.
         /// </summary>
-        public UserAgentBuilder()
+        /// <param name="logger">An optional logging interface</param>
+        public UserAgentBuilder(ILogger logger = null)
         {
+            _logger = logger;
```

Comment:
Here we are defaulting `_logger` to null, but the code below is going to throw a `NullReferenceException` in that case.

Technically you can fix this by using the null coalescing operator everywhere. But it's very easy to forget one place, and most of the time unit tests don't exhaustively cover logging.
```cs
_logger?.LogDebug("An error occurred.");
```

Instead I would recommend initializing `_logger` to a [NullLogger](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.abstractions.nulllogger?view=aspnetcore-2.1) if the `logger` parameter is null.

Reply:
Yep, good point.  I switched it to use an instance of `NullLogger` if needed.